storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/FeatureRowDecoder.java

```diff
@@ -36,60 +41,100 @@ public FeatureRowDecoder(String featureSetRef, FeatureSetSpec spec) {
    *
    * @param featureRow Feature row
    * @return boolean
    */
-  public boolean isEncodingValid(FeatureRow featureRow) {
-    return featureRow.getFieldsList().size() == spec.getFeaturesList().size();
+  private boolean isEncodedV2(FeatureRow featureRow) {
+    return featureRow.getFieldsList().stream().allMatch(field -> !field.getName().isEmpty());
   }
 
   /**
-   * Decoding feature row by repopulating the field names based on the corresponding feature set
-   * spec.
+   * Decode feature row encoded by {@link RedisCustomIO}. NOTE: support for decoding Feature Row v1
+   * encoding will be dropped in Feast 0.7
    *
+   * @throws IllegalArgumentException if unable to the decode the given feature row
    * @param encodedFeatureRow Feature row
    * @return boolean
    */
   public FeatureRow decode(FeatureRow encodedFeatureRow) {
-    final List<Field> fieldsWithoutName = encodedFeatureRow.getFieldsList();
+    if (isEncodedV1(encodedFeatureRow)) {
+      // TODO: remove support for v1 feature row in Feast 0.7
+      final List<Field> fieldsWithoutName = encodedFeatureRow.getFieldsList();
+      List<String> featureNames =
+          spec.getFeaturesList().stream()
+              .sorted(Comparator.comparing(FeatureSpec::getName))
+              .map(FeatureSpec::getName)
+              .collect(Collectors.toList());
 
-    List<String> featureNames =
-        spec.getFeaturesList().stream()
-            .sorted(Comparator.comparing(FeatureSpec::getName))
-            .map(FeatureSpec::getName)
-            .collect(Collectors.toList());
-    List<Field> fields =
-        IntStream.range(0, featureNames.size())
-            .mapToObj(
-                featureNameIndex -> {
-                  String featureName = featureNames.get(featureNameIndex);
-                  return fieldsWithoutName
-                      .get(featureNameIndex)
-                      .toBuilder()
-                      .setName(featureName)
-                      .build();
-                })
-            .collect(Collectors.toList());
-    return encodedFeatureRow
-        .toBuilder()
-        .clearFields()
-        .setFeatureSet(featureSetRef)
-        .addAllFields(fields)
-        .build();
+      List<Field> fields =
+          IntStream.range(0, featureNames.size())
+              .mapToObj(
+                  featureNameIndex -> {
+                    String featureName = featureNames.get(featureNameIndex);
+                    return fieldsWithoutName
+                        .get(featureNameIndex)
+                        .toBuilder()
+                        .setName(featureName)
+                        .build();
+                  })
+              .collect(Collectors.toList());
+
+      return encodedFeatureRow
+          .toBuilder()
+          .clearFields()
+          .setFeatureSet(featureSetRef)
+          .addAllFields(fields)
+          .build();
+
+    } else if (isEncodedV2(encodedFeatureRow)) {
```

Comment:
Do you need an `else` statement here? I think two `if` statements would be cleaner.

Reply:
Updated